### ISSUE TITLE
Add Unused Calculation Items to BPA Rules

### ISSUE THREAD

**Issue by relax528** [TabularEditor/TabularEditor#920]
Sometimes during the development process, calculation groups / items are created but then abandoned. Is it possible to write a BPA rule that could point out items/groups never used. This would keep a model clean.
Maybe classify under [DAX Measures] or a new one for [Calculation Groups] Calculation items not referenced in measures.

**Comment by marcosqlbi** [comment]
How do you identify whether calculation items are not used? They are typically used by reports only (and BPA cannot analyze linked reports).
If used by measures, they should be hidden to the user because a calculation group used in a measure *AND* in the report might create unexpected results (remember that there is no recursion). 
You might want to apply that BPA rule only to hidden calculation groups... is that your use case?

**Comment by relax528** [comment]
Yes, in my case I use them to apply more complicated logic to multiple base measures and they are not used directly in the report. With regards to using in a report, there are other tools like Power BI Documenter that can indicate if it's not used in a report. 

I know these BPA rules are recommendations and if used in the report, they could be ignored. I've inherited a few reports like this and it's not easy to see if they have been referenced.
Thanks

**Comment by didierterrien** [comment]
You might have a look at Power BI cleaner from Imke Feldmann:
[https://www.thebiccountant.com/2021/06/27/power-bi-cleaner-gen2-is-here/](https://www.thebiccountant.com/2021/06/27/power-bi-cleaner-gen2-is-here/)

> **Areas covered**
> The following areas of field usage are covered in these versions:
> - DAX expressions (measures, columns, hierarchies, tables, report level measures, roles, bins, calculation items and title expressions)
> - Incremental refresh policies
> - Relationships
> - Filters (visual-, page- and report level)
> - Visuals (incl. conditional formatting)
